birdset/datamodule/components/transforms.py

```diff
@@ -451,23 +454,59 @@ def _prepare_call(self):
             self.spec_aug = None
             self.nocall_sampler = None
         return
-
-
-class EmbeddingTransforms(BaseTransforms):
-    def __init__(self, task: Literal['multiclass', 'multilabel'] = "multiclass", sampling_rate: int = 3200, max_length: int = 5, decoding: EventDecoding | None = None, feature_extractor: DefaultFeatureExtractor | None = None) -> None:
-        super().__init__(task, sampling_rate, max_length, decoding, feature_extractor)
     
-    def _transform(self, batch):
-        embeddings = [embedding for embedding in batch["embeddings"]]
+class EmbeddingTransforms(BirdSetTransformsWrapper):
+    def __init__(self,
+                task: Literal['multiclass', 'multilabel'] = "multilabel",
+                sampling_rate: int = 32000,
+                model_type: Literal['vision', 'waveform'] = "vision",
+                spectrogram_augmentations: DictConfig = DictConfig({}), # TODO: typing is wrong, can also be List of Augmentations
+                waveform_augmentations: DictConfig = DictConfig({}), # TODO: typing is wrong, can also be List of Augmentations
+                decoding: EventDecoding | None = None,
+                feature_extractor: DefaultFeatureExtractor = DefaultFeatureExtractor(),
+                max_length: int = 5,
+                nocall_sampler: NoCallMixer | None = None, 
+                preprocessing: PreprocessingConfig | None = PreprocessingConfig()) -> None:
+        super().__init__(task, sampling_rate, model_type, spectrogram_augmentations, waveform_augmentations, decoding, feature_extractor, max_length, nocall_sampler, preprocessing)
+    
+    
+    def _get_waveform_batch(self, batch):
```

Comment:
@XgamerTV What do you think of implementing the embedding extraction as a preprocessing step and not as a transform? This would have the following benefits:
- we can save the preprocessed dataset and therefore only have to compute the embeddings once (per experiment or with additional caching once per model). This improves the speed drastically. I would guess that the current implementation is too slow for a bit bigger datasets like HSN.
- do not have to think about batched processing.
- for fine tuning experiments we cannot use the "averaging of the embeddings" anyway (I guess) and will train the models similar as from stretch but with pretrained weights